src/compiler/lowering/SliceColOpLowering.cpp

```diff
@@ -0,0 +1,178 @@
+        : mlir::OpConversionPattern<daphne::SliceColOp>(typeConverter, ctx, PatternBenefit(1)) {
+        this->setDebugName("SliceColOpLowering");
+    }
+
+    /**
+     * @brief Replaces a Transpose operation with a Linalg TransposeOp if possible.
+     *
+     * @return mlir::success if Transpose has been replaced, else mlir::failure.
+     */
+    LogicalResult matchAndRewrite(daphne::SliceColOp op, OpAdaptor adaptor,
+                                  ConversionPatternRewriter &rewriter) const override {
+
+        daphne::MatrixType matrixType = adaptor.getSource().getType().dyn_cast<daphne::MatrixType>();
+        if (!matrixType) {
+            return failure();
+        }
+
+        Location loc = op->getLoc();
+
+        Type matrixElementType = matrixType.getElementType();
+        ssize_t numRows = matrixType.getNumRows();
+        ssize_t numCols = matrixType.getNumCols();
+
+        if (numRows < 0 || numCols < 0) {
+            return rewriter.notifyMatchFailure(
+                op, "sliceColOp codegen currently only works with matrix dimensions that are known at compile time");
+        }
+
+        Value argMemref = rewriter.create<daphne::ConvertDenseMatrixToMemRef>(
+            loc, MemRefType::get({numRows, numCols}, matrixElementType), adaptor.getSource());
+
+        auto lowerIncl = adaptor.getLowerIncl().getDefiningOp<daphne::ConstantOp>().getValue().dyn_cast<mlir::IntegerAttr>().getSInt();
+        auto upperExcl = adaptor.getUpperExcl().getDefiningOp<daphne::ConstantOp>().getValue().dyn_cast<mlir::IntegerAttr>().getSInt();
+        
+        Value resMemref = rewriter.create<memref::AllocOp>(loc, MemRefType::get({numRows, (upperExcl-lowerIncl)}, matrixElementType));
+        
+        DenseI64ArrayAttr offset = rewriter.getDenseI64ArrayAttr({0, lowerIncl});
+        DenseI64ArrayAttr sizes = rewriter.getDenseI64ArrayAttr({numRows, (upperExcl-lowerIncl)});
+        DenseI64ArrayAttr strides = rewriter.getDenseI64ArrayAttr({1, 1});
+        
+        Value selMemref = rewriter.create<memref::SubViewOp>(loc, argMemref, offset, sizes, strides);
+
+        SmallVector<AffineMap, 2> indexMaps{AffineMap::getMultiDimIdentityMap(2, rewriter.getContext()),
+                                            AffineMap::getMultiDimIdentityMap(2, rewriter.getContext())};
+
+        SmallVector<utils::IteratorType, 2> iterTypes{utils::IteratorType::parallel,
+                                                      utils::IteratorType::parallel};
+
+        rewriter.create<linalg::GenericOp>(loc, TypeRange{}, ValueRange{selMemref}, ValueRange{resMemref},
+                                           indexMaps, iterTypes,
+                                           [&](OpBuilder &OpBuilderNested, Location locNested, ValueRange arg) {
+                                               OpBuilderNested.create<linalg::YieldOp>(locNested, arg[0]);
+                                           });
```

Comment:
This shouldn't be needed, as we want to create a view of a matrix we don't want to copy the data. The `memref::SubViewOp` should be enough. You'll need to properly lower it afterwards by adding the following passes to the pipeline:
`mlir::memref::createExpandStridedMetadataPass;` and `mlir::createFinalizeMemRefToLLVMConversionPass`.